packages/admin/dashboard/src/dashboard-app/dashboard-app.tsx

```diff
@@ -196,9 +197,50 @@ export class DashboardApp {
       tempRegistry[item.path] = navItem
     })
 
+    // Sort menu items by rank (ascending order, undefined ranks come last)
+    registry.forEach((items, key) => {
+      const sorted = this.sortMenuItemsByRank(items)
+      registry.set(key, sorted)
+    })
+
     return registry
   }
 
+  private sortMenuItemsByRank(
```

Comment:
should we extract the sorting logic into a util function that is called here and in the test?